### ISSUE TITLE
What's the plan for Intercon sign-ups?

### ISSUE THREAD

**Issue by marleighnorton** [neinteractiveliterature/intercode#2506]
So there was this big town hall last year about changing sign-ups.  I haven't seen any notes about what happened there, but it seems likely the community (certainly I was expecting) some movement on this issue for Intercon T.  

Jaelan should be part of this conversation, since I'm told this is a policy level thing, but I'm not seeing a way to tag them on github.  Dave says he's on it.

Anyway, the whole thing about button mashing at a particular time is really lousy usability for a multitude of reasons.  Based on a conversation with @dkapell , my current thinking is to keep the signups mostly the same except without the button mashing.

- Folks sign up ahead of time for their first choice game.  We list it as "Pending" or something.
- When sign-ups open, we automatically assign folks to their requested games.  If there is overflow, we randomly pick who gets in and who gets on the waitlist and in what order.
- (I think there should be a mitigating feature where for oversubscribed games, people who got waitlisted get first crack at their chosen games in the next round and people who were cast get a minus, but I'm less wedded to that)
- E-mail is sent out telling people what they got, with reminders to sign up for the next round.
- Regardless, people can then modify their signups as they wish.  If they've been waitlisted and would rather be cast in another game, go ahead and do that.

**Comment by dkapell** [comment]
In this suggestion, you're proposing processing signups in click order, is that correct?  That just moves the signup rush to when the con staff makes the schedule available, which isn't at a pre-announced time, which gives con staff or people online at that moment a huge advantage.  Otherwise, if you're going to randomize the signups, this is the Desire Token setup that I documented here: https://docs.google.com/document/d/1cwIZQxYgkIwiSW7e8gVgbJEsbHtm7-qRfHMdMtCAJh8/edit?usp=sharing

**Comment by marleighnorton** [comment]
No.  No click order.

**Comment by marleighnorton** [comment]
Yes, the Desire Token setup—which is a weird way of phrasing it—is basically what I'm in favor of.

**Comment by dkapell** [comment]
To be clear, 'Desire Token' is a development side name for it, which is
meant to describe how it works, and not the final name that would become
user-visible.  I think most of the way that it works would end up invisible
to the the users.  (it solves for the situation where you sign up for the
con after the first round of signups, and then get to directly sign up for
one game, and then request another one, or similar)

**Comment by marleighnorton** [comment]
Yup, that's fine.  If we do that, we can figure out better language, no problem.

**Comment by marleighnorton** [comment]
Welcome @jhartwin !  Share with us your decisions, so we have time to make and test changes!

**Comment by jhartwin** [comment]
So I have a problem. I agree with you that button-mashing is terrible (and in it pretty strongly negatively affects fact some people I care about very much), but feedback from the meetings and discussions we've had so far make it pretty clear that there is a very vocal group of attendees who would strongly object to any changes to signups. I don't want to just let the loudest voices win, but I don't know how to get a good sense of what the majority of the community actually wants, since whenever this comes up, we hear the same handful of voices loudly opposing any change. Also, frankly, at this point I'm scared of the political fallout of making any such change, even if we did find out that the majority of the community wanted it.

**Comment by marleighnorton** [comment]
Do these loud voices have reasons mention the reasons they don't want changes?

**Comment by jhartwin** [comment]
The sense that I've gotten is that for many of them, any system where they can't immediately see whether they got into the game they were going for feels unfair to them. I think it's possible that for at least some of the people who object to the idea of change, once they got used to the new system they'd be fine with it, and some of this is just "ack, change is inherently bad."
(@nbudin or @dkapell, feel free to correct me; you were both involved in these discussions as well)

**Comment by nbudin** [comment]
(Please note that in this comment, I'm explicitly not trying to argue for one side or another of this - I'm writing this to expand on a point @jhartwin alluded to but I think bears a little more clarification.)

To clarify that slightly: the issue I think @jhartwin is mentioning was that in a system where you do some kind of signup request in advance (such as a lottery or @dkapell's Desire Token proposal), there's a period of time between when you take make the request and when you find out whether you got the signup you requested.  That period of time can be anxiety-provoking (and whether or not it's more anxiety-provoking than the anticipation of click-like-mad-o'clock seems to vary a lot from one person to another).  I do have some sympathy for this on a personal level, because I found the waiting period in the Be Con signup system pretty anxiety-provoking.

Whether this is reason enough to not change is very debatable.  On some level I think it boils down to which of the following someone would personally find worse:

* Finding out you didn't get what you wanted immediately
* Not knowing whether or not you got what you wanted for a period of time

**Comment by dkapell** [comment]
Also, because games will fill, you still need to be present at signup time
to learn if you get waitlisted or not, and try to immediately drop from a
waitlist and grab another larp before something else fills.  You don't have
to do that, of course, but you also don't "have" to be present on the site
right at signups under the current system, which is why we give 6 days to
make the selection.

**Comment by nbudin** [comment]
Here, have a presentation file: [town hall mockups.pdf](https://github.com/neinteractiveliterature/intercode/files/3432304/town.hall.mockups.pdf)

This was the slide deck we presented at the signups town hall at Forum@Intercon S, containing some terrible programmer-made mockups that I made for both of the major change proposals we'd been hearing at that point.  (In this deck, "Desire Token" is called "Single Pre-Signup", which is also not a great name but at least doesn't feel vaguely NSFW.)

**Comment by valleyviolet** [comment]
You can design a system that is less likely to leave people on wait lists (like let them pick 3 things they want to be in in priority order and try those in order when their turn comes up) but I don't know if this would really ameliorate any of the concerns.

I suspect that many of the people who are voicing protest are upset not just that there would be change, but that there would be change that would remove the semblance of personal agency they're clinging to in a bad situation. 

(I know I get frustrated because since I do a lot of GMing there's often only one slot and one game I want. If I don't get that I probably won't play that year. But that's an artifact of how I deal with Intercons, not how most people do.)

**Comment by marleighnorton** [comment]
[As a side note, the stuff I'm proposing here is trying to meet the needs of the group as I understand them.  If I were designing the Marleigh Signup System, I'd do some sort of ranked preferences and have the computer assign all the things in one fell swoop, then let folks fiddle with it.  I dislike having the whole thing drag on for weeks.]

I'm going to set aside Dave's reminder that you technically don't have to camp the site at sign-ups.  To have a chance at the popular stuff, functionally you do, so we should take that into account in any design changes.

Many of the comments here make me think we're dealing with brain weasels, which are hard to design for, since by definition they're individuals' hang ups.  Like, I could point out if the anticipation is hard, you can click your desire token at 6:55 for the 7pm roll off and only have to wait five minutes, but somehow I suspect that's not actually helpful to the people having the issues.  

I also suspect people who are having the worst time with the current sign up system are not being heard, because many have already been frustrated into leaving.  As a new Intercon attendee, I was deeply put off by the idea that I had to rearrange my life around a sign up to get a crack at a game I was super excited for.

For Jaelan's PR problem, that's hard.  I strongly suspect the current sign up system is actively hostile to folks with a number of disabilities.  Blindness, reading disorders, attention disorders, and hand-eye coordination impairments come to mind immediately, but I'd want to double check with an expert.  That might be a good way to frame any changes we make, to be more inclusive.  Also it would be true.

**Comment by dkapell** [comment]
I've made the argument that using the desire system at 6:59 is almost the same as the current system before, yeah.  It helps, but not very much.

Having it "drag on for weeks" is actually a benefit for the con, as many GMs only propose their games after they see what they get into in the first round, and we get about 30% of the games each year after first round.  It sucks, but I don't see a way to change that.

Having a single round with ranked preferences means that if you missed entering your preferences before that round, because you heard about the con later, then almost everything is already full.  Also, many people would adust their preferences for later rounds (additional games) based on what they get into in earlier rounds/games. I.e. I like Sci-fi games, and maybe my top 5 games at the con are all sci-fi, but I don't want to play only sci-fi games, so after I get into 2, I'd rather play a historic game as my third signup.

I will reiterate that most games don't fill first round.  Some do, but those would fill first round no matter what, so if there was a lottery to get in, you still wouldn't have a better chance of getting in, and might actually have a less chance, if more people go for it, because they don't need to weigh getting into it vs something else in the moment of signups.  We, the representatives of Intercon, should be very careful when we talk about games filling fast, because it can give the impression that games that didn't fill right away are less good, which can be hurtful to the GMs as well as building on the impression that if you don't get into a very small number of games right at signups, you've lost out on having a good time at the con.

**Comment by valleyviolet** [comment]
If GMs are choosing to hold off on bidding things there are ways you can incentivize them to bid sooner, like giving them a preferential round of sign ups if they bid early. I'm sure any change like that will have it's own political stumbling blocks, but it's possible to attempt to change the way GMs perceive and interact with the system.

**Comment by dkapell** [comment]
Let's assume that preferential signups for any group of people are off the table entirely at this time, for the purposes of this discussion.  It's my firm belief that the community would reject that very strongly.

**Comment by valleyviolet** [comment]
Sure. I intended my point to be less "you should do this example thing" and more "you can design to incentivize GM behavior you want."

**Comment by diewald** [comment]
Hi all, 

I’ve talked to a lot of people about this over the last couple of years, including Brandon Brylawski about his “medical residency hospital matching” system. What follows are some observations I’ve made.

Whenever I’ve asked about something other than button mashing, the biggest concern is that the system will break for two or more friends who want to get into the same game. Or, if you’re the Chicago crowd, you and a dozen of your friends want to be in the same game. Or, in Charley Sumner’s case, getting into the same game as his teenage daughter, to help her if needed. (Or Dan Abraham, or Kreg Segall — we have a lot of next generation LARPers coming with their parents, and more in the pipeline.) There’s comfort for dad or mom to be in game. (And, for me these days, there’s comfort in having my daughter be a GM with me.) 

A delay in knowing the game you’re in prevents some of the legitimate gaming we see in the signup process. X signs up for Game 1, which gets some interest. X’s friend Y signs up for Game 2, which is really popular. X now has a choice twenty minutes into open signups — jump out of Game 1, which looks like it will still have slots in the next round and get into Game 2 before it fills, or stay put and probably not play Game 2. There’s a *lot* of this. 

A surprising number of people are concerned about transparency in the signup system. They know that whenever they make that click, the process that follows is simple. They also know, almost immediately, whether or not they got in. There is comfort in knowing that the system is simple and can’t be (easily) gamed. There are rumors of people who’ve written scripts to hammer the system trying to get in ASAP, and people don’t like that. There is concern that there will be ways to game a new signup system, or that GMs will get special treatment, or that the con staff will be able to get whatever they want. These fears may be unfounded, but they exist, and should have an effect on the UX persona we design for. 

People gripe now about not getting into games, but the transparency of the system means everyone else knows that the system wasn’t at fault. Imagine if a couple of well connected people griped on social media that the “new Intercon signup system is rigged.” We couldn’t conclusively prove that the system isn’t rigged, because it’s complex code, Greek to most. We know how much attention we get when there’s a problem with the site on a signup day. 

I will now return to the work I was doing on the NEIL side of the house — I found what I was looking for here,
Jeff

p.s. Marleigh, this is my first interaction with you AFAIK, so hello! Looong time developer here, long time UX designer / lecturer, last gig as cybersecurity developer / secure code lecturer-teacher.

[Migraine at 29 months, currently a 5 of 10. Test results raise questions. Typos, missing words, and other language oddities may result.]

**Comment by nbudin** [comment]
Re Brandon's residency matching idea, this is (I think) essentially what Be Con does.  They take in everyone's preferences and run several simulations of signups with different random seeds, and their software automatically picks one (or a few) that make everyone theoretically reasonably happy with their schedules.

There are some additional wrinkles to their system that I think are non-obvious, particularly around how they handle safety-related requests not to be placed with certain other attendees, and if other cons were to adopt such a system, they would probably have to come up with their own answers to the safety questions (whether it ends up the same way as Be Con's answer or not).

**Comment by marleighnorton** [comment]
We're starting to run out of time to make any changes on this.  Unless @jhartwin says something soon, I think no decision is no change.

**Comment by jhartwin** [comment]
Honestly I was kind of under the impression that we were already out of time to make this change for T. It's not something I'm comfortable changing unilaterally, as it will affect future Intercons. I've talked with Dave about suggesting that the NEIL Board make a recommendation for the U con chair.

**Comment by marleighnorton** [comment]
Oh look, we have a U con chair.  Can we add Allison to this?

**Comment by marleighnorton** [comment]
Closing this since it's been implemented, so up to individual cons whether they want to use it.
